src/BlockchainData.js

```diff
@@ -0,0 +1,228 @@
+		    if (error) {
+		        console.log(error)
+		    } else {
+		        this.setState({ tokenImperialBalance: balance });
+		       	console.log('The token Imperial balance of account is: ', this.state.tokenImperialBalance);
+		    }
+		})
+
+		this.state.tokenDemocratic.methods.balanceOf(this.state.etherAddress).call({from: this.state.etherAddress}, (error, balance) => {
+		    if (error) {
+		        console.log(error)
+		    } else {
+		        this.setState({ tokenDemocraticBalance: balance });
+		       	console.log('The token Democratic balance of account is: ', this.state.tokenDemocraticBalance);
+		    }
+		})
+
+      	this.state.multisigERC20Token.methods.tokenBalances(this.state.etherAddress, TOKENIMPERIAL_SYMBOL).call({from: this.state.etherAddress}, (error, balance) => {
+			if (error) {
+		        console.log(error)
+		    } else {
+		        this.setState({ tokenImperialContractBalance: balance });
+		       	console.log('The imperial token contract balance of account is: ', this.state.tokenImperialContractBalance);
+		    }
+		})
+
+		this.state.multisigERC20Token.methods.tokenBalances(this.state.etherAddress, TOKENDEMOCRATIC_SYMBOL).call({from: this.state.etherAddress}, (error, balance) => {
+			if (error) {
+		        console.log(error)
+		    } else {
+		        this.setState({ tokenDemocraticContractBalance: balance });
+		       	console.log('The democratic token contract balance of account is: ', this.state.tokenDemocraticContractBalance);
+		    }
+		})
+  	}
+
+  	onHandleClick = (e) => {
+  		e.preventDefault();
+  		var current = document.getElementsByClassName("active");
+		current[0].className = current[0].className.replace(" active", "");
+		e.target.className += " active";
+		const name = e.target.name;
+
+	    switch (name) {
+	    case 'Ethereum': 
+	      if (this.state.Ethereum != true) {
+			this.setState({ 'Ethereum': true });
+	      }
+	      break;
+	    case 'ERC20': 
+	      if (this.state.Ethereum == true) {
+			this.setState({ 'Ethereum': false });
+	      }
+	      break;   
+	    default:
+	      break;
+	    }
+  	}
+  	
+	render() {
+  	  	return (
+      		<div id="content" className ="smart">
+      			<div className="btnGroup">
+				  	<button className="buttonInGroup active" name="Ethereum" onClick={this.onHandleClick}>Ethereum</button>
+				  	<button className="buttonInGroup" name="ERC20" onClick={this.onHandleClick}>ERC20</button>
+      			</div>
+      			{
+      			this.state.Ethereum ? 
+        		<SmartContract 
+        			address={this.props.etherAddress} 
+        			etherBalance={this.state.etherBalance}
+        			contractBalance={this.state.contractBalance}
+        			updateBalances={this.updateBalances}
+        			web3={this.state.web3}
+        			multisig={this.state.multisig} 
+        			multisigAddress={MULTISIG_ADDRESS}/>
+        		:
+        		<ERC20 
+        			web3={this.state.web3}
```

Comment:
nearly all of these can be changed just to 
```
<ERC20 
..state
/>
```

in that case you will pass whole state and props names will be same as state names

also you can do
```
<ERC20 
..state
tokenImperialBalance={25}
name="raid"
/>
```
to add or override something that you passed by **...state**